Lib/fractions.py

```diff
@@ -244,7 +243,8 @@ def __new__(cls, numerator=0, denominator=None):
                 self._denominator = numerator.denominator
                 return self
 
-            elif isinstance(numerator, (float, Decimal)):
+            elif (isinstance(numerator, numbers.Number) and
```

Comment:
I cannot comment on the TypeError below but you should update:

```python
raise TypeError("argument should be a string or a Rational instance")
```

since you do not need a rational only (https://github.com/python/cpython/pull/120271/files#diff-f561eb7eb97f832b2698837f52c2c2cf23bdb0b31c69cf1f6aaa560280993316R281-R282).